pkg/webui/console/containers/gateway-location-form/gateway-location-form.js

```diff
@@ -0,0 +1,181 @@
+        latitudeRegexp.test(String(value)),
+      ),
+    otherwise: schema => schema.strip(),
+  }),
+  longitude: Yup.number().when('update_location_from_status', {
+    is: false,
+    then: schema =>
+      schema.test('is-valid-longitude', sharedMessages.validateLatLong, value =>
+        longitudeRegexp.test(String(value)),
+      ),
+    otherwise: schema => schema.strip(),
+  }),
+  altitude: Yup.number().when('update_location_from_status', {
+    is: false,
+    then: schema => schema.integer(sharedMessages.validateInt32).required(),
+    otherwise: schema => schema.strip(),
+  }),
+  location_public: Yup.bool(),
+  update_location_from_status: Yup.bool(),
+})
+
+const getRegistryLocation = function(antennas) {
+  let registryLocation
+  if (antennas) {
+    for (const key of Object.keys(antennas)) {
+      if (antennas[key].location.source === 'SOURCE_REGISTRY') {
+        registryLocation = { antenna: antennas[key], key }
+        break
+      }
+    }
+  }
+  return registryLocation
+}
+
+const GatewayLocationForm = ({ gateway, gatewayId, updateGateway }) => {
+  const registryLocation = getRegistryLocation(gateway.antennas)
+  const initialValues = {
+    location_public: gateway.location_public || false,
+    update_location_from_status: gateway.update_location_from_status || false,
+    ...(registryLocation
+      ? registryLocation.antenna.location
+      : {
+          latitude: undefined,
+          longitude: undefined,
+          altitude: undefined,
+        }),
+  }
+
+  const [updateLocationFromStatus, setUpdateLocationFromStatus] = useState(
+    initialValues.update_location_from_status,
+  )
+
+  const handleSubmit = useCallback(
+    async values => {
+      const patch = {
+        location_public: values.location_public,
+        update_location_from_status: values.update_location_from_status,
+      }
+      if (!values.update_location_from_status) {
+        const registryLocation = getRegistryLocation(gateway.antennas)
+        if (registryLocation) {
+          // Update old location value
+          patch.antennas = [...gateway.antennas]
+          patch.antennas[registryLocation.key].location = {
+            ...registryLocation.antenna.location,
+            ...values,
+          }
+        } else {
+          // Create new location value
+          patch.antennas = [
+            {
+              gain: 0,
+              location: {
+                ...values,
+                accuracy: 0,
+                source: 'SOURCE_REGISTRY',
+              },
+            },
+          ]
+        }
+      }
+
+      await updateGateway(gatewayId, patch)
+    },
+    [gateway, gatewayId, updateGateway],
+  )
+
+  const handleDelete = useCallback(async () => {
+    const registryLocation = getRegistryLocation(gateway.antennas)
+
+    const patch = {
+      antennas: [...gateway.antennas],
+    }
+    patch.antennas.splice(registryLocation.key, 1)
+
+    await updateGateway(gatewayId, patch)
```

Comment:
the caller is responsible for waiting for the response if necessary. no reason to do it here
```suggestion
    return updateGateway(gatewayId, patch)
```